### ISSUE TITLE
Authenticate to storage account using Azure username rather than SAS key?

### ISSUE THREAD

**Issue by jonnyry** [microsoft/AzureTRE#3927]
Using Storage Explorer running on a VM within the TRE, is it possible (/should it be possible) to connect to a storage account (such as the workspace's storage account) using Azure Entra ID credentials, rather than using SAS keys?

**Comment by tim-allen-ck** [comment]
It isn't possible currently, this issue here [#1148](https://github.com/microsoft/AzureTRE/issues/1148) could help if this was to become a feature.

**Comment by jonnyry** [comment]
Ah, I'm not suggesting that the user is logged into the Windows VM using their Azure credetials (although that would be great)...  Just that within a VM, signing into Storage Explorer using Azure credentials.

I've got as far as signing in the browser but then Storage Explorer returns an a sign in error:

![image](https://github.com/microsoft/AzureTRE/assets/7916532/0bf955e2-2f13-4ea5-91d2-595baa955020)

I presume the firewall is blocking outbound connections however not sure of the implications of opening the firewall for this endpoint, and was wondering whether you'd considered this in the past?

**Comment by tim-allen-ck** [comment]
Yes you're right, currently the firewall is blocking the signin url. We could have this as configuration option? Some users may not want to open the firewall up.
